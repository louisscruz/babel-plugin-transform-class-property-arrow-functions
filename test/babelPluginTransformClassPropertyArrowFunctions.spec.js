import pluginTester from 'babel-plugin-tester';

import babelPluginTransformClassPropertyArrowFunctions from '../src';

pluginTester({
  plugin: babelPluginTransformClassPropertyArrowFunctions,
  tests: [
    {
      title:
        'it handles scenarios where a class property without parameters assigned to a function returning an empty block is provided in a class with a constructor',
      code: `
        class A {
          constructor() {}

          a = () => {};
        }
      `,
      snapshot: true,
    },
    {
      title:
        'it handles scenarios where a class property without parameters assigned to a function returning an empty block is provided in a class without a constructor',
      code: `
        class A {
          a = () => {};
        }
      `,
      snapshot: true,
    },
    {
      title:
        'it handles scenarios where a class property with a single parameter assigned to a function returning an empty block is provided in a class without a constructor',
      code: `
        class A {
          a = parameterOne => {};
        }
      `,
      snapshot: true,
    },
    {
      title:
        'it handles scenarios where a class property with multiple parameters assigned to a function returning an empty block is provided in a class without a constructor',
      code: `
        class A {
          a = (parameterOne, parameterTwo) => {};
        }
      `,
      snapshot: true,
    },
    {
      title:
        'it handles scenarios where a class property without parameters assigned to a function returning an implicit return is provided in a class with a constructor',
      code: `
        class A {
          constructor() {}

          a = () => 1;
        }
      `,
      snapshot: true,
    },
    {
      title:
        'it handles scenarios where a class property without parameters assigned to a function returning an implicit return is provided in a class without a constructor',
      code: `
        class A {
          a = () => 1;
        }
      `,
      snapshot: true,
    },
    {
      title:
        'it handles scenarios where a class property with a single parameter assigned to a function returning an implicit return is provided in a class without a constructor',
      code: `
        class A {
          a = parameterOne => parameterOne;
        }
      `,
      snapshot: true,
    },
    {
      title:
        'it handles scenarios where a class property with multiple parameters assigned to a function returning an implicit return is provided in a class without a constructor',
      code: `
        class A {
          a = (parameterOne, parameterTwo) => parameterTwo;
        }
      `,
      snapshot: true,
    },
    {
      title:
        'it handles scenarios where a class property without parameters assigned to a function returning a void-returning block is provided in a class with a constructor',
      code: `
        class A {
          constructor() {}

          a = () => {
            console.log('hi');
          };
        }
      `,
      snapshot: true,
    },
    {
      title:
        'it handles scenarios where a class property without parameters assigned to a function returning a void-returning block is provided in a class without a constructor',
      code: `
        class A {
          a = () => {
            console.log('hi');
          };
        }
      `,
      snapshot: true,
    },
    {
      title:
        'it handles scenarios where a class property with a single parameter assigned to a function returning a void-returning block is provided in a class without a constructor',
      code: `
        class A {
          a = parameterOne => {
            console.log('hi');
          };
        }
      `,
      snapshot: true,
    },
    {
      title:
        'it handles scenarios where a class property with multiple parameters assigned to a function returning a void-returning block is provided in a class without a constructor',
      code: `
        class A {
          a = (parameterOne, parameterTwo) => {
            console.log('hi');
          };
        }
      `,
      snapshot: true,
    },
    {
      title:
        'it handles scenarios where a class property without parameters assigned to a function returning an explicit return is provided in a class with a constructor',
      code: `
        class A {
          constructor() {}

          a = () => {
            const value = 1 + 2;
            return value;
          };
        }
      `,
      snapshot: true,
    },
    {
      title:
        'it handles scenarios where a class property without parameters assigned to a function returning an explicit return is provided in a class without a constructor',
      code: `
        class A {
          a = () => {
            const value = 1 + 2;
            return value;
          };
        }
      `,
      snapshot: true,
    },
    {
      title:
        'it handles scenarios where a class property with a single parameter assigned to a function returning an explicit return is provided in a class without a constructor',
      code: `
        class A {
          a = parameterOne => {
            const value = 1 + 2;
            return value;
          };
        }
      `,
      snapshot: true,
    },
    {
      title:
        'it handles scenarios where a class property with multiple parameters assigned to a function returning an explicit return is provided in a class without a constructor',
      code: `
        class A {
          a = (parameterOne, parameterTwo) => {
            const value = 1 + 2;
            return value;
          };
        }
      `,
      snapshot: true,
    },
    {
      title:
        'it handles scenarios where a class property without parameters assigned to a function returning an explicit return is provided in a class with a constructor',
      code: `
        class A {
          constructor() {}

          a = () => {
            const value = 1 + 2;
            return value;
          };
        }
      `,
      snapshot: true,
    },
    {
      title:
        'it handles scenarios where a class property with a single parameter assigned to a function returning an explicit return is provided in a class with a constructor',
      code: `
        class A {
          constructor() {}

          a = parameterOne => {
            const value = 1 + 2;
            return value;
          };
        }
      `,
      snapshot: true,
    },
    {
      title:
        'it handles scenarios where a class property with multiple parameters assigned to a function returning an explicit return is provided in a class with a constructor',
      code: `
        class A {
          constructor() {}

          a = (parameterOne, parameterTwo) => {
            const value = 1 + 2;
            return value;
          };
        }
      `,
      snapshot: true,
    },
  ],
});
