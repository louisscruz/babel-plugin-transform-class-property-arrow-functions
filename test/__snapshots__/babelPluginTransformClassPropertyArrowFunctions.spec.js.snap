// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`babelPluginTransformClassPropertyArrowFunctions it handles scenarios where a class property with a single parameter assigned to a function returning a void-returning block is provided in a class without a constructor: it handles scenarios where a class property with a single parameter assigned to a function returning a void-returning block is provided in a class without a constructor 1`] = `
"
class A {
  a = parameterOne => {
    console.log('hi');
  };
}

      ↓ ↓ ↓ ↓ ↓ ↓

\\"use strict\\";

class A {
  constructor() {
    this.a = this.a.bind(this);
  }

  a(parameterOne) {
    console.log('hi');
  }

}
"
`;

exports[`babelPluginTransformClassPropertyArrowFunctions it handles scenarios where a class property with a single parameter assigned to a function returning an empty block is provided in a class without a constructor: it handles scenarios where a class property with a single parameter assigned to a function returning an empty block is provided in a class without a constructor 1`] = `
"
class A {
  a = parameterOne => {};
}

      ↓ ↓ ↓ ↓ ↓ ↓

\\"use strict\\";

class A {
  constructor() {
    this.a = this.a.bind(this);
  }

  a(parameterOne) {}

}
"
`;

exports[`babelPluginTransformClassPropertyArrowFunctions it handles scenarios where a class property with a single parameter assigned to a function returning an explicit return is provided in a class with a constructor: it handles scenarios where a class property with a single parameter assigned to a function returning an explicit return is provided in a class with a constructor 1`] = `
"
class A {
  constructor() {}

  a = parameterOne => {
    const value = 1 + 2;
    return value;
  };
}

      ↓ ↓ ↓ ↓ ↓ ↓

\\"use strict\\";

class A {
  constructor() {
    this.a = this.a.bind(this);
  }

  a(parameterOne) {
    const value = 1 + 2;
    return value;
  }

}
"
`;

exports[`babelPluginTransformClassPropertyArrowFunctions it handles scenarios where a class property with a single parameter assigned to a function returning an explicit return is provided in a class without a constructor: it handles scenarios where a class property with a single parameter assigned to a function returning an explicit return is provided in a class without a constructor 1`] = `
"
class A {
  a = parameterOne => {
    const value = 1 + 2;
    return value;
  };
}

      ↓ ↓ ↓ ↓ ↓ ↓

\\"use strict\\";

class A {
  constructor() {
    this.a = this.a.bind(this);
  }

  a(parameterOne) {
    const value = 1 + 2;
    return value;
  }

}
"
`;

exports[`babelPluginTransformClassPropertyArrowFunctions it handles scenarios where a class property with a single parameter assigned to a function returning an implicit return is provided in a class without a constructor: it handles scenarios where a class property with a single parameter assigned to a function returning an implicit return is provided in a class without a constructor 1`] = `
"
class A {
  a = parameterOne => parameterOne;
}

      ↓ ↓ ↓ ↓ ↓ ↓

\\"use strict\\";

class A {
  constructor() {
    this.a = this.a.bind(this);
  }

  a(parameterOne) {
    return parameterOne;
  }

}
"
`;

exports[`babelPluginTransformClassPropertyArrowFunctions it handles scenarios where a class property with multiple parameters assigned to a function returning a void-returning block is provided in a class without a constructor: it handles scenarios where a class property with multiple parameters assigned to a function returning a void-returning block is provided in a class without a constructor 1`] = `
"
class A {
  a = (parameterOne, parameterTwo) => {
    console.log('hi');
  };
}

      ↓ ↓ ↓ ↓ ↓ ↓

\\"use strict\\";

class A {
  constructor() {
    this.a = this.a.bind(this);
  }

  a(parameterOne, parameterTwo) {
    console.log('hi');
  }

}
"
`;

exports[`babelPluginTransformClassPropertyArrowFunctions it handles scenarios where a class property with multiple parameters assigned to a function returning an empty block is provided in a class without a constructor: it handles scenarios where a class property with multiple parameters assigned to a function returning an empty block is provided in a class without a constructor 1`] = `
"
class A {
  a = (parameterOne, parameterTwo) => {};
}

      ↓ ↓ ↓ ↓ ↓ ↓

\\"use strict\\";

class A {
  constructor() {
    this.a = this.a.bind(this);
  }

  a(parameterOne, parameterTwo) {}

}
"
`;

exports[`babelPluginTransformClassPropertyArrowFunctions it handles scenarios where a class property with multiple parameters assigned to a function returning an explicit return is provided in a class with a constructor: it handles scenarios where a class property with multiple parameters assigned to a function returning an explicit return is provided in a class with a constructor 1`] = `
"
class A {
  constructor() {}

  a = (parameterOne, parameterTwo) => {
    const value = 1 + 2;
    return value;
  };
}

      ↓ ↓ ↓ ↓ ↓ ↓

\\"use strict\\";

class A {
  constructor() {
    this.a = this.a.bind(this);
  }

  a(parameterOne, parameterTwo) {
    const value = 1 + 2;
    return value;
  }

}
"
`;

exports[`babelPluginTransformClassPropertyArrowFunctions it handles scenarios where a class property with multiple parameters assigned to a function returning an explicit return is provided in a class without a constructor: it handles scenarios where a class property with multiple parameters assigned to a function returning an explicit return is provided in a class without a constructor 1`] = `
"
class A {
  a = (parameterOne, parameterTwo) => {
    const value = 1 + 2;
    return value;
  };
}

      ↓ ↓ ↓ ↓ ↓ ↓

\\"use strict\\";

class A {
  constructor() {
    this.a = this.a.bind(this);
  }

  a(parameterOne, parameterTwo) {
    const value = 1 + 2;
    return value;
  }

}
"
`;

exports[`babelPluginTransformClassPropertyArrowFunctions it handles scenarios where a class property with multiple parameters assigned to a function returning an implicit return is provided in a class without a constructor: it handles scenarios where a class property with multiple parameters assigned to a function returning an implicit return is provided in a class without a constructor 1`] = `
"
class A {
  a = (parameterOne, parameterTwo) => parameterTwo;
}

      ↓ ↓ ↓ ↓ ↓ ↓

\\"use strict\\";

class A {
  constructor() {
    this.a = this.a.bind(this);
  }

  a(parameterOne, parameterTwo) {
    return parameterTwo;
  }

}
"
`;

exports[`babelPluginTransformClassPropertyArrowFunctions it handles scenarios where a class property without parameters assigned to a function returning a void-returning block is provided in a class with a constructor: it handles scenarios where a class property without parameters assigned to a function returning a void-returning block is provided in a class with a constructor 1`] = `
"
class A {
  constructor() {}

  a = () => {
    console.log('hi');
  };
}

      ↓ ↓ ↓ ↓ ↓ ↓

\\"use strict\\";

class A {
  constructor() {
    this.a = this.a.bind(this);
  }

  a() {
    console.log('hi');
  }

}
"
`;

exports[`babelPluginTransformClassPropertyArrowFunctions it handles scenarios where a class property without parameters assigned to a function returning a void-returning block is provided in a class without a constructor: it handles scenarios where a class property without parameters assigned to a function returning a void-returning block is provided in a class without a constructor 1`] = `
"
class A {
  a = () => {
    console.log('hi');
  };
}

      ↓ ↓ ↓ ↓ ↓ ↓

\\"use strict\\";

class A {
  constructor() {
    this.a = this.a.bind(this);
  }

  a() {
    console.log('hi');
  }

}
"
`;

exports[`babelPluginTransformClassPropertyArrowFunctions it handles scenarios where a class property without parameters assigned to a function returning an empty block is provided in a class with a constructor: it handles scenarios where a class property without parameters assigned to a function returning an empty block is provided in a class with a constructor 1`] = `
"
class A {
  constructor() {}

  a = () => {};
}

      ↓ ↓ ↓ ↓ ↓ ↓

\\"use strict\\";

class A {
  constructor() {
    this.a = this.a.bind(this);
  }

  a() {}

}
"
`;

exports[`babelPluginTransformClassPropertyArrowFunctions it handles scenarios where a class property without parameters assigned to a function returning an empty block is provided in a class without a constructor: it handles scenarios where a class property without parameters assigned to a function returning an empty block is provided in a class without a constructor 1`] = `
"
class A {
  a = () => {};
}

      ↓ ↓ ↓ ↓ ↓ ↓

\\"use strict\\";

class A {
  constructor() {
    this.a = this.a.bind(this);
  }

  a() {}

}
"
`;

exports[`babelPluginTransformClassPropertyArrowFunctions it handles scenarios where a class property without parameters assigned to a function returning an explicit return is provided in a class with a constructor: it handles scenarios where a class property without parameters assigned to a function returning an explicit return is provided in a class with a constructor 1`] = `
"
class A {
  constructor() {}

  a = () => {
    const value = 1 + 2;
    return value;
  };
}

      ↓ ↓ ↓ ↓ ↓ ↓

\\"use strict\\";

class A {
  constructor() {
    this.a = this.a.bind(this);
  }

  a() {
    const value = 1 + 2;
    return value;
  }

}
"
`;

exports[`babelPluginTransformClassPropertyArrowFunctions it handles scenarios where a class property without parameters assigned to a function returning an explicit return is provided in a class with a constructor: it handles scenarios where a class property without parameters assigned to a function returning an explicit return is provided in a class with a constructor 2`] = `
"
class A {
  constructor() {}

  a = () => {
    const value = 1 + 2;
    return value;
  };
}

      ↓ ↓ ↓ ↓ ↓ ↓

\\"use strict\\";

class A {
  constructor() {
    this.a = this.a.bind(this);
  }

  a() {
    const value = 1 + 2;
    return value;
  }

}
"
`;

exports[`babelPluginTransformClassPropertyArrowFunctions it handles scenarios where a class property without parameters assigned to a function returning an explicit return is provided in a class without a constructor: it handles scenarios where a class property without parameters assigned to a function returning an explicit return is provided in a class without a constructor 1`] = `
"
class A {
  a = () => {
    const value = 1 + 2;
    return value;
  };
}

      ↓ ↓ ↓ ↓ ↓ ↓

\\"use strict\\";

class A {
  constructor() {
    this.a = this.a.bind(this);
  }

  a() {
    const value = 1 + 2;
    return value;
  }

}
"
`;

exports[`babelPluginTransformClassPropertyArrowFunctions it handles scenarios where a class property without parameters assigned to a function returning an implicit return is provided in a class with a constructor: it handles scenarios where a class property without parameters assigned to a function returning an implicit return is provided in a class with a constructor 1`] = `
"
class A {
  constructor() {}

  a = () => 1;
}

      ↓ ↓ ↓ ↓ ↓ ↓

\\"use strict\\";

class A {
  constructor() {
    this.a = this.a.bind(this);
  }

  a() {
    return 1;
  }

}
"
`;

exports[`babelPluginTransformClassPropertyArrowFunctions it handles scenarios where a class property without parameters assigned to a function returning an implicit return is provided in a class without a constructor: it handles scenarios where a class property without parameters assigned to a function returning an implicit return is provided in a class without a constructor 1`] = `
"
class A {
  a = () => 1;
}

      ↓ ↓ ↓ ↓ ↓ ↓

\\"use strict\\";

class A {
  constructor() {
    this.a = this.a.bind(this);
  }

  a() {
    return 1;
  }

}
"
`;
